Formula/convox.rb

```diff
@@ -19,21 +20,22 @@ class Convox < Formula
   end
 
   def install
-    ENV["GOPATH"] = buildpath
+    ENV["GOPATH"] = buildpath/"go"
 
-    (buildpath/"src/github.com/convox/rack").install Dir["*"]
+    (buildpath/"src").install Dir["*"]
 
     resource("packr").stage { system "go", "install", "./packr" }
-    cd buildpath/"src/github.com/convox/rack" do
-      system buildpath/"bin/packr"
+
+    cd buildpath/"src" do
+      system "../go/bin/packr"
+      system "go", "build", "-mod=vendor", "-ldflags=-X main.version=#{version}",
+             "-o", bin/"convox", "-v", "./cmd/convox"
     end
 
-    system "go", "build", "-ldflags=-X main.version=#{version}",
-           "-o", bin/"convox", "-v", "github.com/convox/rack/cmd/convox"
     prefix.install_metafiles
   end
 
   test do
-    system bin/"convox"
+    assert_equal version, shell_output("#{bin}/convox -v").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.